added detailed GSSoC'24 contribution

maha-sachin opened the pull request:
This issue fixes Add GSSOC Contribution Details to Contribution Guide #91

## Added GSSOC Contribution Details to Contribution Guide

### Description
The contribution guide currently lacks details about how GSSOC'24 participants can contribute to the project effectively. It's important to provide clear instructions and guidelines to help GSSOC participants get started with contributing.

### Instructions
- Add information about GSSOC labels (`gssoc`, `level-1`, `level-2`, `level-3`) to the contribution guide.
- Provide clear instructions on how GSSOC participants can find tasks that match their skills and interests.
- Include any specific guidelines or tips for GSSOC participants to ensure a smooth contribution process.

Here is the updation screenshot:

<img width="1722" alt="Screenshot 2024-05-15 at 4 02 01 PM" src="https://github.com/ashutoshgithubs/iemhacks_DEV-G/assets/122064139/a83d6cb0-0b77-40eb-9136-df3a22fa6995">


- This helps GSSOC'24 participants often look for tasks matching their skill levels. Labeling issues with level-1, level-2, level-3 allows participants to easily identify tasks that suit their expertise, fostering a positive contribution experience.

- Increased Contributions: By making it easier for GSSOC participants to contribute, you're likely to receive more contributions during the program period, leading to project growth and improvement.

maha-sachin committed:
added detailed GSSoC'24 contribution

maha-sachin commented:
Hi @ashutoshgithubs , Can you review this PR and I would appreciate your feedback.

ashutoshgithubs reviewed (commented):
Hi @maha-sachin , This PR is good to merge. Before that you have to modify the levels nomenclature as per the instruction of the program manager. For example :
`level-1` should be` level1` and so on. `gssoc` tag is right. Please do the needful.

Commit:
Update contribution level details

corrected contribution level details.
changed level-1 to level1, level-2 to level2 and level-3 to level3.

maha-sachin commented:
@ashutoshgithubs,I've made some updates to the level details and would appreciate it if you could take a look when you have a moment.
